storagemarket/impl/provider_states.go

```diff
@@ -45,13 +46,13 @@ func (p *Provider) validating(ctx context.Context, deal MinerDeal) (func(*MinerD
 	if err != nil {
 		return nil, err
 	}
-	if head.Height() >= deal.Proposal.ProposalExpiration {
+	if head.Height() >= deal.Proposal.StartEpoch {
```

Comment:
Not really related to this PR, but this check doesn't give us any slack time to seal/commit a sector before the `StartEpoch`.

Reply:
yea we should make decision about that in the future